### PR TITLE
[ENG-2435] Upgrade pytest and others to fix broken tests

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

This is a booster shot of updated decencies for the OSF for pytest, these such keep us more up to date with the current test infrastructure and hopefully fix some probabilistic test fails.

You can read about the bug [here](https://github.com/pytest-dev/pytest/issues/3171) which indicates updating pytest will fix the issue. I picked pytest 5.0.0 because it is recent and only requires few changes. The biggest change is pytest no long allows passing of fixtures the same way. [Read about it here](https://github.com/pytest-dev/pytest/issues/3950). So that had to be changed to release this. 

## Changes

- update pytest and related dependencies
- turn fixtures into functions to appease pytest update
 

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that travis tests pass as expected

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Side Effects

None that I know of. 

## Ticket

https://openscience.atlassian.net/browse/ENG-2435

### CODE REVIEW

File: tests/test_registrations/test_review_flows.py [hunk 1]
```diff
@@ -110,10 +133,10 @@ def test_rejection_flow(self, sanction_fixture, initial_state, end_state):
         assert registration.moderation_state == end_state.db_name
 
 
-    @pytest.mark.parametrize('sanction_fixture', [registration_approval, embargo, retraction])
-    def test_approve_after_reject_fails(self, sanction_fixture):
+    @pytest.mark.parametrize('sanction_object', [registration_approval, embargo, retraction])
+    def test_approve_after_reject_fails(self, sanction_object):
         # using fixtures in parametrize returns the function
```
Comment (jwalz) [inline tests/test_registrations/test_review_flows.py:138]: nit: This comment is now unnecessary/misleading (in all locations)

**Comment by jwalz** [comment]
Can you explain in the description why updating pytest fixed the test linked in the ticket? Also, more detail on what, specifically, pytest doesn't like about these fixtures now (seemingly it doesn't support using fixtures that aren't fully evaluated in test setup, a la via parametrize).

**Comment by Johnetordoff** [comment]
@jwalz ok I made some comments in the description

File: tests/test_registrations/test_review_flows.py [hunk 2]
```diff
@@ -135,7 +135,6 @@ def test_rejection_flow(self, sanction_object, initial_state, end_state):
 
     @pytest.mark.parametrize('sanction_object', [registration_approval, embargo, retraction])
     def test_approve_after_reject_fails(self, sanction_object):
-        # using fixtures in parametrize returns the function
```
Comment (jwalz) [inline tests/test_registrations/test_review_flows.py:115]: It's not a huge deal, but if you could go through the file and delete all iterations of this comment, that would be appreciated

**Review by jwalz — APPROVED** [review]

**Pull request merged**